velox/exec/tests/TableEvolutionFuzzerTest.cpp

```diff
@@ -0,0 +1,650 @@
+        case TypeKind::BIGINT:
+          return liftToPrimitiveType<int64_t>(*typed, type);
+        default:
+          VELOX_UNREACHABLE();
+      }
+    }
+    case TypeKind::REAL: {
+      auto* typed = input->asChecked<FlatVector<float>>();
+      switch (type->kind()) {
+        case TypeKind::REAL:
+          return input;
+        case TypeKind::DOUBLE:
+          return liftToPrimitiveType<double>(*typed, type);
+        default:
+          VELOX_UNREACHABLE();
+      }
+    }
+    case TypeKind::ARRAY: {
+      VELOX_CHECK_EQ(type->kind(), TypeKind::ARRAY);
+      auto* array = input->asChecked<ArrayVector>();
+      return std::make_shared<ArrayVector>(
+          config_.leafPool,
+          type,
+          array->nulls(),
+          array->size(),
+          array->offsets(),
+          array->sizes(),
+          liftToType(array->elements(), type->asArray().elementType()));
+    }
+    case TypeKind::MAP: {
+      VELOX_CHECK_EQ(type->kind(), TypeKind::MAP);
+      auto& mapType = type->asMap();
+      auto* map = input->asChecked<MapVector>();
+      return std::make_shared<MapVector>(
+          config_.leafPool,
+          type,
+          map->nulls(),
+          map->size(),
+          map->offsets(),
+          map->sizes(),
+          liftToType(map->mapKeys(), mapType.keyType()),
+          liftToType(map->mapValues(), mapType.valueType()));
+    }
+    case TypeKind::ROW: {
+      VELOX_CHECK_EQ(type->kind(), TypeKind::ROW);
+      auto& rowType = type->asRow();
+      auto* row = input->asChecked<RowVector>();
+      auto children = row->children();
+      for (int i = 0; i < rowType.size(); ++i) {
+        auto& childType = rowType.childAt(i);
+        if (i < children.size()) {
+          children[i] = liftToType(children[i], childType);
+        } else {
+          children.push_back(BaseVector::createNullConstant(
+              childType, row->size(), config_.leafPool));
+        }
+      }
+      return std::make_shared<RowVector>(
+          config_.leafPool,
+          type,
+          row->nulls(),
+          row->size(),
+          std::move(children));
+    }
+    default:
+      return input;
+  }
+}
+
+void TableEvolutionFuzzer::run() {
+  std::vector<column_index_t> bucketColumnIndices;
+  for (int j = 0; j < config_.columnCount; ++j) {
+    if (folly::Random::oneIn(2 * config_.columnCount, rng_)) {
+      bucketColumnIndices.push_back(j);
+    }
+  }
+  VLOG(1) << "bucketColumnIndices: [" << folly::join(", ", bucketColumnIndices)
+          << "]";
+  auto setups = makeSetups(bucketColumnIndices);
+  auto tableDir = TempDirectoryPath::create();
+  std::vector<std::unique_ptr<TaskCursor>> writeTasks(
+      2 * config_.evolutionCount - 1);
+  for (int i = 0; i < config_.evolutionCount; ++i) {
+    auto data = vectorFuzzer_.fuzzRow(setups[i].schema, kVectorSize, false);
+    auto actualDir = fmt::format("{}/actual_{}", tableDir->getPath(), i);
+    VELOX_CHECK(std::filesystem::create_directory(actualDir));
```

Comment:
why not use velox fs to create?

Reply:
Just one fewer dependency since testing that component is not in the scope of this fuzzer